### PR TITLE
[FIX] website: perf - reduce SQL queries on hot execution on website page

### PR DESCRIPTION
**Pull request by rdeodoo**
This PR is the continuity of #46925

It is a bunch of small commits, retro compatible, that should reduce
the number of sql query for each page.

--

Optimistic stats
```
|                       | Query |   Time |
| --------------------- | ----- | ------ |
| website.page          |       |        |
| Before                |    25 | 0.0218 |
| Now                   |    12 | 0.0106 |
| tracked website.page  |       |        |
| Before                |    34 | 0.0276 |
| Now                   |    20 | 0.0123 |
| tracked homepage      |       |        |
| Before                |    34 | 0.0223 |
| Now                   |    19 | 0.0194 |
```

task-2211013

### CODE REVIEW

File: addons/website/models/website.py [hunk 1]
```diff
@@ -126,10 +127,16 @@ def _onchange_language_ids(self):
...
             self.default_lang_id = language_ids[0]
 
+    @api.depends('menu_ids.parent_id')
     def _compute_menu(self):
-        Menu = self.env['website.menu']
+        # accessing through menu_ids allows to prefetch all menus at once
+        self.menu_ids.child_id
         for website in self:
-            website.menu_id = Menu.search([('parent_id', '=', False), ('website_id', '=', website.id)], order='id', limit=1).id
+            website.menu_id = False
```
Comment (Feyensv) [inline addons/website/models/website.py]: the default assignation can be done once before the for loop ;)
`self.menu_id = False`

Comment (rdeodoo) [reply addons/website/models/website.py]: This is still a POC, the whole code got replaced, but thanks

**Comment by rdeodoo** [comment]
@rco-odoo I think this is what you suggested. Indeed it reduce the query number from 46 to 33 just for the menus

File: addons/website/models/website.py [hunk 2]
```diff
@@ -127,9 +127,16 @@ def _onchange_language_ids(self):
             self.default_lang_id = language_ids[0]
 
     def _compute_menu(self):
-        Menu = self.env['website.menu']
         for website in self:
-            website.menu_id = Menu.search([('parent_id', '=', False), ('website_id', '=', website.id)], order='id', limit=1).id
+            menu_ids = self.env['website.menu'].search([('website_id', '=', website.id)])
+            for menu_id in menu_ids:
```
Comment (rco-odoo) [inline addons/website/models/website.py]: `menu_ids` → `menus`
`menu_id` → `menu`

Comment (rco-odoo) [reply addons/website/models/website.py]: comment: "use field parent_id (1 query) to determine field child_id (2 queries by level)"

File: odoo/addons/base/models/ir_translation.py [hunk 3]
```diff
@@ -237,9 +239,11 @@ def _modified(self):
                         record = model.browse(trans.res_id)
                         record.modified([field.name])
         for trans in self:
-            if trans.type != 'model' or trans.name.split(',')[0] in self.CACHED_MODELS:
-                self.clear_caches()
-                break
+            if trans.type != 'model' \
+                or trans.name.split(',')[0] in self.CACHED_MODELS \
+                    or (self.comments and 'openerp-web' in self.comments):  # static js/xml file - clear get_web_translations_hash
```
Comment (rco-odoo) [inline odoo/addons/base/models/ir_translation.py]:
```suggestion
            if (
                trans.type != 'model'
                or trans.name.split(',')[0] in self.CACHED_MODELS
                or (self.comments and 'openerp-web' in self.comments  # static js/xml file - clear get_web_translations_hash
            ):
```

File: odoo/addons/base/models/ir_translation.py [hunk 4]
```diff
@@ -897,3 +908,14 @@ def get_translations_for_webclient(self, mods, lang):
...
         return translations_per_module, lang_params
+
+    @api.model
+    @tools.ormcache('frozenset(mods)', 'lang')
+    def get_web_translations_hash(self, mods, lang):
+        translations, lang_params = self.get_translations_for_webclient(mods, lang)
+        translation_cache = {
+            'lang_parameters': lang_params,
+            'modules': translations,
+            'lang': lang,
```
Comment (rco-odoo) [inline odoo/addons/base/models/ir_translation.py]:
```suggestion
            'lang': lang,
            'multi_lang': len(request.env['res.lang'].sudo().get_installed()) > 1,
```

File: addons/http_routing/controllers/main.py [hunk 5]
```diff
@@ -8,7 +8,7 @@
 
 class Routing(Home):
 
-    @http.route('/website/translations/<string:unique>', type='http', auth="public", website=True)
+    @http.route('/website/translations/<string:unique>', type='http', auth="public", website=True, multilang=False)
     def get_website_translations(self, unique, lang, mods=None):
```
Comment (rco-odoo) [inline addons/http_routing/controllers/main.py]: JKE check lang

**Comment by rdeodoo** [comment]
Being tested on prod, will be merged in a few days

**Comment by JKE-be** [comment]
@robodoo r+ rebase-merge

**Pull request closed without merging**